src/ApparentHorizons/ChangeCenterOfStrahlkorper.cpp

```diff
@@ -0,0 +1,158 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ApparentHorizons/ChangeCenterOfStrahlkorper.hpp"
+
+#include <cmath>
+
+#include "ApparentHorizons/Strahlkorper.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "NumericalAlgorithms/RootFinding/TOMS748.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace change_expansion_center_of_strahlkorper_detail {
+
+// Get the r_hat vector of the Strahlkorper, which defines the
+// collocation points.
+template <typename Frame>
+tnsr::i<DataVector, 3, Frame> get_rhat(
+    const Strahlkorper<Frame>& strahlkorper) noexcept {
+  const auto new_theta_phi = strahlkorper.ylm_spherepack().theta_phi_points();
```

Comment:
Is there a reason to not use a const reference, to avoid the allocation?

Reply:
You get an allocation no matter what, since theta_phi_points() does an allocation and returns an object.

Using a const reference would work too (C++ guarantees that a reference to a temporary remains valid during the scope of that reference), but would save nothing because of return value optimization, and also would be slightly misleading.

Reply:
Ah right, it's the explicit-ness that gives issues in return types :)